redfish-core/lib/pcie.hpp

```diff
@@ -91,6 +94,204 @@ static inline void
+        return;
+    }
+}
+
+/**
+ * @brief Get PCIeSlot properties.
+ * @param[in,out]   asyncResp       Async HTTP response.
+ * @param[in]       pcieSlotPath    Object path of the PCIeSlot.
+ * @param[in]       serviceMap      A map to hold Service and corresponding
+ * interface list for the given cable id.
+ */
+static inline void
+    getPcieSlotLinkStatus(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const std::string& pcieSlotPath,
+                          const dbus::utility::MapperServiceMap& serviceMap)
+{
+    for (const auto& [service, interfaces] : serviceMap)
+    {
+        for (const auto& interface : interfaces)
+        {
+            if (interface != "xyz.openbmc_project.Inventory.Item.PCIeSlot")
+            {
+                continue;
+            }
+
+            crow::connections::systemBus->async_method_call(
+                [asyncResp](const boost::system::error_code ec,
+                            const dbus::utility::DbusVariantType& property) {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    const std::string* value =
+                        std::get_if<std::string>(&property);
+                    if (value == nullptr)
+                    {
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    std::string linkStatus = *value;
+                    if (!linkStatus.empty())
+                    {
+                        fillPcieDeviceStatus(asyncResp->res, linkStatus);
+                        return;
+                    }
+                },
+                service, pcieSlotPath, "org.freedesktop.DBus.Properties", "Get",
+                interface, "LinkStatus");
+        }
+    }
+}
+
+/**
+ * @brief Get subtree map for PCIeSlots.
+ * @param[in,out]   asyncResp       Async HTTP response.
+ * @param[in]       pcieSlotPath    Object path of the PCIeSlot.
+ * @param[in]       pcieDevice      PCIe device name/ID.
+ */
+static inline void
+    getPcieSlotSubTree(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       const std::string& pcieSlotPath,
+                       const std::string& pcieDevice)
+{
+    auto respHandler = [asyncResp, pcieSlotPath,
+                        pcieDevice](const boost::system::error_code ec,
+                                    const MapperGetSubTreeResponse& subTree) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "DBUS response error on GetSubTree"
+                             << ec.message();
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        if (subTree.size() == 0)
+        {
+            BMCWEB_LOG_ERROR << "Can't find PCIeSlot D-Bus object!";
```

Comment:
I guess no error here is fine.. If you don't have PCIeSlot, you don't get the Link Status